From e569d6d1499287698b6a9eed6ee99bc624e075c6 Mon Sep 17 00:00:00 2001
From: xu zhang <xuzhang4@microsoft.com>
Date: Wed, 04 Sep 2024 06:59:18 +0000
Subject: [PATCH] Check nullptr to avoid crash.

Execution_context could be a null pointer, which would lead to a null
pointer access when calling dom_window->GetFrame(), causing the renderer
process to crash.
Therefore, add a check here to avoid a crash.

Bug: 361461518
Change-Id: I3aa51f5f12341fe193dc4ab9e653e78ab377e786
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5816708
Reviewed-by: Yuki Shiino <yukishiino@chromium.org>
Commit-Queue: xu ms <xuzhang4@microsoft.com>
Reviewed-by: David Baron <dbaron@chromium.org>
Reviewed-by: Dave Tapuska <dtapuska@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1350622}
---

--- a/third_party/blink/renderer/core/dom/element.cc
+++ b/third_party/blink/renderer/core/dom/element.cc
@@ -9366,6 +9366,12 @@
 void Element::LogAddElementIfIsolatedWorldAndInDocument(
     const char element[],
     const QualifiedName& attr1) {
+  // TODO(crbug.com/361461518): Investigate the root cause of execution context
+  // is unexpectedly null.
+  if (!GetDocument().GetExecutionContext()) {
+    return;
+  }
+
   if (!isConnected() ||
       !V8DOMActivityLogger::HasActivityLoggerInIsolatedWorlds()) {
     return;
